pkg/controller/core/workload_controller.go

```diff
@@ -346,6 +346,9 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 	if err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true); err != nil {
 		return false, client.IgnoreNotFound(err)
 	}
+	if err := workload.SetAllChecksToPending(ctx, r.client, wl); err != nil {
```

Comment:
Actually, I think it is better to update the state for the checks just after `workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByAdmissionCheck, message)`, and submit in the `ApplyAdmissionStatus`, because now it requires to Patches, and both update the status.

Reply:
Agree, fixed